src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingSubmissionResource.java

```diff
@@ -51,15 +60,34 @@ public ProgrammingSubmissionResource(ProgrammingSubmissionService programmingSub
      */
     @PostMapping(value = Constants.PROGRAMMING_SUBMISSION_RESOURCE_PATH + "{participationId}")
     public ResponseEntity<?> notifyPush(@PathVariable("participationId") Long participationId, @RequestBody Object requestBody) {
-
         log.info("REST request to inform about new commit+push for participation: {}", participationId);
-        programmingSubmissionService.notifyPush(participationId, requestBody);
+
+        try {
+            ProgrammingSubmission submission = programmingSubmissionService.notifyPush(participationId, requestBody);
+            // notify the user via websocket.
+            messagingTemplate.convertAndSend("/topic/participation/" + participationId + "/newSubmission", submission);
+        }
+        catch (IllegalArgumentException ex) {
+            log.error("Exception encountered when trying to extract the commit hash from the request body. Processing submission for participation {} with request object {}: {}",
```

Comment:
I would prefer 

> Processing submission for participation {} **failed** with request object {}: {}

Reply:
Why do you want to make the error logging more generic instead of giving the indication about what happened most likely?

Reply:
For me, the second sentence in the log message sounded more like an alternative approach ('something went wrong and now we try to process it in order to solve it') rather than 'something failed and this is the cause', but this is probably a minor thing and not that important for me tbh.

Reply:
Also I did not want to remove the first sentence, just add the 'failed' into the second sentence, I should have marked that better.